charts/nfs-subdir-external-provisioner/templates/storageclass.yaml

```
@@ -3,14 +3,14 @@ apiVersion: storage.k8s.io/v1
 kind: StorageClass
 metadata:
   labels:
-    app: {{ template "nfs-subdir-external-provisioner.name" . }}
-    chart: {{ template "nfs-subdir-external-provisioner.chart" . }}
-    heritage: {{ .Release.Service }}
-    release: {{ .Release.Name }}
+    {{- include "nfs-subdir-external-provisioner.labels" . | nindent 4 }}
   name: {{ .Values.storageClass.name }}
 {{- if .Values.storageClass.defaultClass }}
   annotations:
     storageclass.kubernetes.io/is-default-class: "true"
+  {{- with .Values.storageClass.annotations }}
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
 {{- end }}
```

Your change does not take any effect here when this storage class isn't the defaultClass :-)
```suggestion
  annotations:
  {{- if .Values.storageClass.defaultClass }}
    storageclass.kubernetes.io/is-default-class: "true"
  {{- end }}
  {{- with .Values.storageClass.annotations }}
    {{- toYaml . | nindent 4 }}
  {{- end }}
```